src/app/features/Chanels/redux/actions.js

```diff
@@ -33,9 +44,45 @@ export function fetchPosts(id) {
       requestPosts(id).then((result) => {
         const postItem = result.find((item) => item.channelId === id);
         if (postItem) {
-          dispatch(getPosts(postItem));
+          dispatch(getPostsAction(postItem));
         }
       });
     }
   };
 }
+
+export function addChannel(channelName) {
+  return async (dispatch, getState) => {
+    const channelExists = getState().channelsState.find(
+      (channel) => channel.name === channelName
+    );
+
+    if (!channelExists) {
+      requestChannel(channelName).then((result) => {
```

Comment:
please, rename this function to **createChannel** and encapsulate localStorage logic in it. Do NOT mix localStorage logic and redux logic.

Reply:
the **createChannel** should return newly created channel item.